weave-js/src/components/Button/Button.tsx

```diff
@@ -4,153 +4,153 @@
  * https://www.figma.com/file/01KWBdMZg5QM9SRS1pQq0z/Design-System----Robot-Styles?type=design&node-id=5956-31813&mode=design&t=Gm4WWGWwgjdfUUTe-0
  */
 
-import {TooltipContentProps} from '@radix-ui/react-tooltip';
+import {type TooltipContentProps} from '@radix-ui/react-tooltip';
 import classNames from 'classnames';
-import React, {ReactElement, useState} from 'react';
+import React, {
+  CSSProperties,
+  forwardRef,
+  type ReactElement,
+  useState,
+} from 'react';
 import {twMerge} from 'tailwind-merge';
 
-import {Icon, IconName} from '../Icon';
+import {Icon, type IconName} from '../Icon';
 import * as Tooltip from '../RadixTooltip';
 import {Tailwind} from '../Tailwind';
-import {ButtonSize, ButtonVariant} from './types';
+import type {ButtonSize, ButtonVariant} from './types';
 
-export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
-  /**
-   * Shorthand alias for `startIcon` since most button icons are startIcons.
-   * If both props are provided, `startIcon` value will be used out of the two.
-   */
+type IconButtonProps = {
```

Comment:
These changes add the following constraints at compile time:
- `tooltip` must be present if using `icon`
- `icon` mustn't be used with `children`, `startIcon`, or `endIcon`
- `children` is expected if not using `icon`

Reply:
It probably varies as the `startIcon` and `icon` were originally interchangeable, but either way we'd need to update the call sites.